### PR TITLE
Support for aggregation names with dots in first element path of a pipeline aggregation

### PR DESCRIPTION
**Pull request by iverase**
When the first element on the path of a pipeline aggregation refers to an aggregation with a name containing a dot, the aggregation fails as it does not save properly the aggregation name during validation. This is due that the validation it's own parsing strategy that cannot deal with such names.

This PR changes the strategy during validation is it uses `AggregationPath` to resolve the name.

### CODE REVIEW

File: server/src/internalClusterTest/java/org/elasticsearch/search/aggregations/pipeline/AvgBucketIT.java [hunk 1]
```diff
@@ -45,6 +45,9 @@
     static int numValueBuckets;
     static long[] valueCounts;
 
+    static String histoName;
+    static String termsName;
+
```
Comment (iverase) [inline server/src/internalClusterTest/java/org/elasticsearch/search/aggregations/pipeline/AvgBucketIT.java:50]: I have only do it for one test but maybe we should do it for all of them for consistency? Maybe a follow up PR?

File: server/src/internalClusterTest/java/org/elasticsearch/search/aggregations/pipeline/AvgBucketIT.java [hunk 2]
```diff
@@ -86,21 +89,29 @@ public void setupSuiteScopeCluster() throws Exception {
...
         indexRandom(true, builders);
         ensureSearchable();
+        histoName = randomName();
+        termsName = randomName();
+    }
+
+    private static String randomName() {
+        return randomBoolean()
+            ? randomAlphaOfLengthBetween(3, 12)
+            : randomAlphaOfLengthBetween(3, 6) + "." + randomAlphaOfLengthBetween(3, 6);
```
Comment (not-napoleon) [inline server/src/internalClusterTest/java/org/elasticsearch/search/aggregations/pipeline/AvgBucketIT.java:99]: We should test non-ascii characters too, probably in a follow up PR.  Or just open a ticket for it, could be a good first issue for someone.

File: server/src/internalClusterTest/java/org/elasticsearch/search/aggregations/pipeline/AvgBucketIT.java (hunk 1 above)
Comment (not-napoleon) [reply server/src/internalClusterTest/java/org/elasticsearch/search/aggregations/pipeline/AvgBucketIT.java:50]: Yeah, I think that'd be good to do in a follow up PR.

**Review by not-napoleon — APPROVED** [review]
LGTM

**Pull request merged**